### PR TITLE
Addon-Test: Implement Global Error Modal

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Relates https://github.com/storybookjs/storybook/issues/29312

## What I did

Implemented the Global Error Modal for the Testing Addon and changed slightly the background css of the Modal component.

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createSize |  0 B | 0 B | 0 B | - | - |
| generateSize |  78.7 MB | 78.7 MB | 148 B | 1.2 | 0% |
| initSize |  152 MB | 152 MB | 488 B | 0.27 | 0% |
| diffSize |  73.7 MB | 73.7 MB | 340 B | -0.48 | 0% |
| buildSize |  7.07 MB | 7.07 MB | -142 B | **-1.41** | 0% |
| buildSbAddonsSize |  1.78 MB | 1.78 MB | 0 B | **-1.4** | 0% |
| buildSbCommonSize |  195 kB | 195 kB | 0 B | - | 0% |
| buildSbManagerSize |  1.85 MB | 1.85 MB | -71 B | **-1.41** | 0% |
| buildSbPreviewSize |  271 kB | 271 kB | 0 B | **-1.41** | 0% |
| buildStaticSize |  0 B | 0 B | 0 B | - | - |
| buildPrebuildSize |  4.09 MB | 4.09 MB | -71 B | **-1.41** | 0% |
| buildPreviewSize |  2.98 MB | 2.97 MB | -71 B | 0.35 | 0% |
| testBuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbAddonsSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbCommonSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbManagerSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| testBuildStaticSize |  0 B | 0 B | 0 B | - | - |
| testBuildPrebuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildPreviewSize |  0 B | 0 B | 0 B | - | - |

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createTime |  14.9s | 20.9s | 5.9s | 0.81 | 28.5% |
| generateTime |  20.9s | 18.7s | -2s -160ms | **-1.86** | 🔰-11.5% |
| initTime |  16.2s | 13.2s | -3s -71ms | **-2.25** | 🔰-23.2% |
| buildTime |  9.4s | 8.2s | -1s -170ms | **-1.59** | 🔰-14.2% |
| testBuildTime |  0ms | 0ms | 0ms | - | - |
| devPreviewResponsive |  6.8s | 7.1s | 338ms | 0.45 | 4.7% |
| devManagerResponsive |  4.3s | 4.3s | -24ms | 0.02 | -0.5% |
| devManagerHeaderVisible |  575ms | 793ms | 218ms | **2.82** | 🔺27.5% |
| devManagerIndexVisible |  613ms | 828ms | 215ms | **2.59** | 🔺26% |
| devStoryVisibleUncached |  737ms | 1.2s | 472ms | 0.02 | 39% |
| devStoryVisible |  610ms | 829ms | 219ms | **2.65** | 🔺26.4% |
| devAutodocsVisible |  532ms | 784ms | 252ms | **2.9** | 🔺32.1% |
| devMDXVisible |  492ms | 662ms | 170ms | **1.82** | 🔺25.7% |
| buildManagerHeaderVisible |  543ms | 555ms | 12ms | -0.09 | 2.2% |
| buildManagerIndexVisible |  550ms | 562ms | 12ms | -0.28 | 2.1% |
| buildStoryVisible |  580ms | 625ms | 45ms | 0.38 | 7.2% |
| buildAutodocsVisible |  515ms | 465ms | -50ms | -1.12 | -10.8% |
| buildMDXVisible |  460ms | 508ms | 48ms | -0.11 | 9.4% |

## Greptile Summary

This PR implements a Global Error Modal for the Testing Addon and updates the Modal component's stories and styles. Here are the key changes:

- Added new `GlobalErrorModal` component in `/code/addons/test/src/components/GlobalErrorModal.tsx`
- Created corresponding stories for `GlobalErrorModal` in `/code/addons/test/src/components/GlobalErrorModal.stories.tsx`
- Updated Modal stories in `/code/core/src/components/components/Modal/Modal.stories.tsx` with improved visual presentation
- Modified Modal styling in `/code/core/src/components/components/Modal/Modal.styled.tsx` for better error handling
- Added `@storybook/theming` as a dependency in `/code/addons/test/package.json`

### CODE REVIEW

File: code/addons/test/src/components/GlobalErrorModal.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  fontFamily: theme.typography.fonts.mono,
+  fontSize: '12px',
+  borderTop: `1px solid ${theme.appBorderColor}`,
+}));
+
+const TroubleshootLink = styled.a(({ theme }) => ({
+  color: theme.color.defaultText,
+}));
+
+const TROUBLESHOOT_LINK = `https://storybook.js.org/docs/vitest/troubleshooting`;
```
Comment (yannbf) [inline code/addons/test/src/components/GlobalErrorModal.tsx:48]: You can use this solution to figure out the correct link which includes version and renderer:
https://github.com/storybookjs/storybook/pull/29294/files#diff-b5d7f930a7ae284cca8c7b8e4dbb1cee16770206bc9cf8f1dd73f8fe6d80fd34R39-R44

And to make the stories work, you can use this:
https://github.com/storybookjs/storybook/pull/29294/files#diff-b92a3cddfcd19801bbd0ebdcb31bfdeb9348d3819bc11de65c9470b63cec5306R40

**Review by yannbf — APPROVED** [review]
LGTM! Please merge the latest into your branch, there might be a small amount of conflicts

File: code/addons/test/src/components/GlobalErrorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+const ModalStackTrace = styled.pre(({ theme }) => ({
+  whiteSpace: 'pre-wrap',
+  overflow: 'auto',
+  maxHeight: '60vh',
+  margin: 0,
+  padding: `20px`,
+  fontFamily: theme.typography.fonts.mono,
+  fontSize: '12px',
+  borderTop: `1px solid ${theme.appBorderColor}`,
```
Comment (yannbf) [inline code/addons/test/src/components/GlobalErrorModal.tsx:44]: seems like the border is round?
image

Comment (valentinpalkovic) [reply code/addons/test/src/components/GlobalErrorModal.tsx:44]: Thanks!

**Pull request merged**